collectors/osv/collectors.py

```diff
@@ -217,12 +223,25 @@ def get_refs(data: dict) -> list:
                 }
             ]
             for ref in data.get("references", []):
-                refs.append(
-                    {
-                        "type": FlawReference.FlawReferenceType.EXTERNAL,
-                        "url": ref["url"],
-                    }
-                )
+                # URLs are validated before saving, and some handling is necessary
+                try:
+                    # URL scheme can be missing, so try to fix it first
+                    if not ref["url"].startswith("http") and not ref["url"].startswith(
+                        "ftp"
+                    ):
+                        ref["url"] = f"http://{ref['url']}"
+
+                    validate = URLValidator()
+                    validate(ref["url"])
+                    refs.append(
+                        {
+                            "type": FlawReference.FlawReferenceType.EXTERNAL,
+                            "url": ref["url"],
+                        }
+                    )
+                except ValidationError:
+                    # Ignore the URL if it is invalid (e.g. due to a typo)
+                    pass
```

Comment:
And this one too. We have to handle invalid links somehow. One option is to ignore them, and the second is to try to fix them (as I did). Some form of handling should also be added to NVD.

Reply:
It's debatable based on what the underlying business process needs. I'd argue that keeping as much original information from the original report (snippet) as possible will help the analyst more than changing or hiding information. Even an invalid URL might be useful for the analyst when used with [Google, Kagi, DuckDuckGo, Bing], and arguably more so than trying to be clever and changing/hiding information.

Reply:
The issue is that `url` in `FlawReference` is `URLField`, so a validation must pass. We are not able to save an invalid link (we could disable validations, but I would be against this option).

Reply:
I agree too.